src/app/modules/sample-providers/components/submission-form/submission-form.component.ts

```diff
@@ -1,22 +1,74 @@
-import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import {
+  Component,
+  OnInit,
+  Output,
+  EventEmitter,
+  Input,
+  ElementRef,
+  ViewChild,
+  HostListener,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+} from '@angular/core';
 import {
   AbstractControl,
   FormBuilder,
   FormGroup,
   Validators,
+  NG_VALUE_ACCESSOR,
+  ControlValueAccessor,
 } from '@angular/forms';
 
 @Component({
   selector: 'app-submission-form',
   templateUrl: './submission-form.component.html',
   styleUrls: ['./submission-form.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: SubmissionFormComponent,
+      multi: true,
+    },
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class SubmissionFormComponent implements OnInit {
+export class SubmissionFormComponent implements OnInit, ControlValueAccessor {
   form: FormGroup;
 
+  f: File; // demo file
```

Comment:
Same comment as above here. Also, it may not be even be necessary to have this variable defined on the class since you can access it off of the event.